solutions/USACO feb16-milk-pails.mdx

```diff
@@ -2,15 +2,96 @@
 id: usaco-620
 source: USACO Silver 2016 February
 title: Milk Pails
-author: Óscar Garries
+author: Óscar Garries, Neo Wang
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_pails_silver_feb16.html)
 
 <LanguageSection>
 <CPPSection>
 
-## C++ Implementation
+## C++ Implementation (BFS)
+
+We can directly simulate all operations using BFS.
+
+In this problem, we only care about states $(i, j)$ where pail $X$ has $i$ and pail $Y$ has $j$ units of milk. We can perform a BFS starting from base state $(0, 0)$ to compute $\texttt{dist}[i][j]$, the minimum number of steps to reach each state. Afterwards, compute the answer by iterating over all states where $\texttt{dist}[i][j] \le K$ and finding the minimum $|i+j-M|$.
```

Comment:
```suggestion
In this problem, we care about states $(i, j)$ where pail $X$ has $i$ and pail $Y$ has $j$ units of milk. We can perform a BFS starting from base state $(0, 0)$ to compute $\texttt{dist}[i][j]$, the minimum number of steps to reach each state. Afterwards, compute the answer by iterating over all states where $\texttt{dist}[i][j] \le K$ and finding the minimum $|i+j-M|$.
```
(yea my original wording wasn't perfect, and it still isn't, but since there's another comment anyways I might as well add some small edits to make it a bit better)